cmd/httpapi/main.go

```diff
@@ -0,0 +1,223 @@
+	"github.com/project-flotta/flotta-operator/internal/repository/edgedeviceset"
+	"github.com/project-flotta/flotta-operator/internal/repository/edgedevicesignedrequest"
+	"github.com/project-flotta/flotta-operator/internal/repository/edgeworkload"
+	"github.com/project-flotta/flotta-operator/internal/storage"
+	"github.com/project-flotta/flotta-operator/internal/yggdrasil"
+	"github.com/project-flotta/flotta-operator/pkg/mtls"
+	"github.com/project-flotta/flotta-operator/restapi"
+	"github.com/project-flotta/flotta-operator/restapi/operations"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/record"
+	"net/http"
+	"os"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	initialDeviceNamespace = "default"
+)
+
+var (
+	operatorNamespace = "flotta"
+	scheme            = runtime.NewScheme()
+)
+
+var Config struct {
+
+	// The port of the HTTPs server
+	HttpsPort uint16 `envconfig:"HTTPS_PORT" default:"8043"`
+
+	// Domain where TLS certificate listen.
+	// FIXME check default here
+	Domain string `envconfig:"DOMAIN" default:"project-flotta.io"`
+
+	// If TLS server certificates should work on 127.0.0.1
+	TLSLocalhostEnabled bool `envconfig:"TLS_LOCALHOST_ENABLED" default:"true"`
+
+	// The address the metric endpoint binds to.
+	MetricsAddr string `envconfig:"METRICS_ADDR" default:":8080"`
+
+	// The address the probe endpoint binds to.
+	ProbeAddr string `envconfig:"PROBE_ADDR" default:":8081"`
+
+	// Verbosity of the logger.
+	LogLevel string `envconfig:"LOG_LEVEL" default:"info"`
+
+	// Client Certificate expiration time
+	ClientCertExpirationTime uint `envconfig:"CLIENT_CERT_EXPIRATION_DAYS" default:"30"`
+
+	// Kubeconfig specifies path to a kubeconfig file if the server is run outside of a cluster
+	Kubeconfig string `envconfig:"KUBECONFIG" default:""`
+}
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+
+	utilruntime.Must(managementv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(obv1.AddToScheme(scheme))
+	utilruntime.Must(routev1.AddToScheme(scheme))
+}
+
+func main() {
+	log, _ := zap.NewProduction()
+	log.WithOptions()
+	defer func() {
+		_ = log.Sync()
+	}()
+	logger := log.Sugar()
+
+	err := envconfig.Process("", &Config)
+	if err != nil {
+		logger.Errorf("Cannot process config: %v", err)
+		panic(err.Error())
+	}
+	clientConfig, err := getRestConfig(Config.Kubeconfig)
+	if err != nil {
+		logger.Errorf("Cannot prepare k8s client config: %v. Kubeconfig was: %s", err, Config.Kubeconfig)
+		panic(err.Error())
+	}
+
+	clnt, err := client.New(clientConfig, client.Options{Scheme: scheme})
```

Comment:
with https://github.com/project-flotta/flotta-operator/pull/201 merged, it can be used to access flotta's crs and maintain local cache using informers.

Reply:
Yes. I think I will do this one in a separate PR.